app/client/components/DetailView.js

```diff
@@ -75,18 +76,37 @@ function DetailView(gristDoc, viewSectionModel) {
 
   //--------------------------------------------------
   // Set up DOM event handling.
+  this.lastElemSelected = null;
 
   // Clicking on a detail field selects that field.
   this.onEvent(this.viewPane, 'mousedown', '.g_record_detail_el', function(elem, event) {
     this.viewSection.hasFocus(true);
     var rowModel = this.recordLayout.getContainingRow(elem, this.viewPane);
     var field = this.recordLayout.getContainingField(elem, this.viewPane);
     commands.allCommands.setCursor.run(rowModel, field);
+    this.lastElemSelected = elem;
   });
 
   // Double-clicking on a field also starts editing the field.
   this.onEvent(this.viewPane, 'dblclick', '.g_record_detail_el', function(elem, event) {
     this.activateEditorAtCursor();
+    this.lastElemSelected = elem;
+  });
+
+  // Clicking on a field also starts editing the field but only on certain cases and if the field are preselected
+  const simpleClickUnacceptedSelectors = [".formula_field", ".widget_checkbox", ".widget_switch"]
```

Comment:
Instead of checking against class names, we can check if a field is a formula column or a toggle column of type checkbox or switch. There's an example in another comment of how to get a reference to the field, from which you can access more information (see `ViewFieldRec.ts`).

Since we're checking for a few different things on fields, it may be helpful to wrap it up into a private method, say `_canSingleClick(field: ViewFieldRec)`, and call it by passing in the clicked element's field. You can then use it in the event handler below once you've confirmed that the clicked field is the same one as the last clicked field.

Reply:
thank, i think it's cleaner indeed